EdgeAgentSDK/Domain/Sources/Models/Message+Codable.swift

```diff
@@ -49,7 +50,12 @@ extension Message: Codable {
     public init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: CodingKeys.self)
         let id = try container.decode(String.self, forKey: .id)
-        let piuri = try container.decode(String.self, forKey: .piuri)
+        let piuri: String
+        if let aux = try container.decodeIfPresent(String.self, forKey: .piuri) {
+            piuri = aux
+        } else {
+            piuri = try container.decode(String.self, forKey: .type)
+        }
```

Comment:
thinking this logic should be inverted? 
where we should try to use `type` first and fallback on `piuri` ?

```suggestion
        if let aux = try container.decodeIfPresent(String.self, forKey: .type) {
            piuri = aux
        } else {
            piuri = try container.decode(String.self, forKey: .piuri)
        }
```

Reply:
I actually removed completely the if. Its It should always be `type`.